src/ansys/mapdl/core/errors.py

```diff
@@ -306,26 +307,73 @@ def wrapper(*args, **kwargs):
                 old_handler = signal.signal(signal.SIGINT, handler)
 
         # Capture gRPC exceptions
-        try:
-            out = func(*args, **kwargs)
-        except grpc.RpcError as error:
-            # Custom errors
-            if error.code() == grpc.StatusCode.RESOURCE_EXHAUSTED:
-                if "Received message larger than max" in error.details():
-                    try:
-                        lim_ = int(error.details().split("(")[1].split("vs")[0])
-                    except IndexError:
-                        lim_ = int(512 * 1024**2)
-
-                    raise MapdlgRPCError(
-                        f"RESOURCE_EXHAUSTED: {error.details()}. "
-                        "You can try to increase the gRPC message length size using 'PYMAPDL_MAX_MESSAGE_LENGTH'"
-                        " environment variable. For instance:\n\n"
-                        f"$ export PYMAPDL_MAX_MESSAGE_LENGTH={lim_}"
+        n_attempts = 3
+        initial_backoff = 0.05
+        multiplier_backoff = 3
+
+        i_attemps = 0
+
+        while True:
+            try:
+                out = func(*args, **kwargs)
+
+                # Exit while-loop if success
+                break
+
+            except grpc.RpcError as error:
+
+                mapdl = retrieve_mapdl_from_args(args)
+
+                i_attemps += 1
+                if i_attemps <= n_attempts:
+
+                    wait = (
+                        initial_backoff * multiplier_backoff**i_attemps
+                    )  # Exponential backoff
+                    sleep(wait)
+
+                    # reconnect
+                    mapdl._log.debug(
+                        f"Re-connection attempt {i_attemps} after waiting {wait:0.3f} seconds"
+                    )
+
+                    connected = mapdl._connect(timeout=wait)
+
+                    # Retry again
+                    continue
+
+                # Custom errors
+                reason = ""
+                suggestion = ""
```

Comment:
These are always going to be empty, right? At least whenever you call the wrapper function

Reply:
Yes. It is just preparing the field for future custom reasons and exceptions. Because ideally, I should be able to catch the gRPC exceptions and translate them to actionable insights (``reason`` and ``suggestion``). That is why those vars.